src/SWP/Bundle/ContentBundle/EventListener/ArticleMetadataSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use SWP\Bundle\ContentBundle\ArticleEvents;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Component\Bridge\Model\PackageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ArticleMetadataSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ArticleEvents::PRE_CREATE => 'populateMetadata',
+        ];
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function populateMetadata(ArticleEvent $event)
+    {
+        $package = $this->getPackage($event);
+        $article = $event->getArticle();
+
+        $article->setMetadata([
```

Comment:
We store it as a copy of the content that came from Superdesk and the Package is independent from Article. It will be used to preview which data came from Superdesk etc. and that's it. That's why we persist a Package.

I don't have anything against database normalization, I just say it's not needed in this case.

Even if we could get rid off of persisting Package we would need to store the metadata in Article object at the end. So what you describing is not needed at all. To me it looks like over-engineering in this case imo but I might be wrong.